core/crypto/client_tcert_pool.go

```diff
@@ -23,7 +23,9 @@ type tCertPool interface {
 
 	Stop() error
 
-	GetNextTCert() (tCert, error)
+	GetNextTCerts(nCerts int) ([]tCert, error)
```

Comment:
p.s. I, too, believe in a (physical and logical) separation using interfaces (re: the `crypto` package) - some of this logic, should not be there, in the first place.